data_structures/binary_tree/segment_tree.py

```diff
@@ -53,19 +54,7 @@ def query_recursive(self, idx, l, r, a, b): #query(1, 1, N, a, b) for query max
 
     def showData(self):
         showList = []
-        for i in range(1,N+1):
+        for i in range(1,(self.N)+1):
```

Comment:
Unecessary parentheses
```suggestion
        for i in range(1,self.N+1):
```